File: competition/First Half 2021/Feb 28. 2021/Solution5689.java
import java.util.*;

class Solution5689 {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
    	int result = 0, index = 0;
    	if(ruleKey.equals("color")){
    		index = 1;
    	}
    	if(ruleKey.equals("name")){
    		index = 2;
    	}
    	for(List<String> ele: items){
    		if(ele.get(index).equals(ruleValue)){
    			result++;
    		}
    	}
    	return result;
    }
}

File: competition/First Half 2021/Feb 28. 2021/Solution5689.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution5689 {
public:
    int countMatches(vector<vector<string>>& items, string ruleKey, string ruleValue) {
        int index = 0;
        if(ruleKey == "color"){
            index = 1;
        }
        else if(ruleKey == "name"){
            index = 2;
        }
        int res = 0;
        for(auto ele: items){
            if(ele[index] == ruleValue){
                res++;
            }
        }
        return res;
    }
};